### DISCUSSION TITLE
Question: use maven wrapper in Eclipse

### DISCUSSION THREAD

**Discussion by tivervac** [eclipse-m2e/m2e-core#1273] [Q&A] [answered]
Related to my recent [PR](https://github.com/eclipse-m2e/m2e-core/pull/1270) I wonder whether it's possible to use maven wrapper in Eclipse - instead of m2e's embedded maven - to handle _all_ maven operations?

Some internet resources seem to say that you can use maven wrapper, but that dependency resolution will always happen through the embedded maven e.g. https://stackoverflow.com/questions/35607335/update-version-of-maven-within-eclipse

**Comment by laeubi** [comment, marked as answer]
> I wonder whether it's possible to use maven wrapper in Eclipse - instead of m2e's embedded maven - to handle all maven operations?

No.

> Some internet resources seem to say that you can use maven wrapper, but that dependency resolution will always happen through the embedded maven

That's not whats said there. Technically it is impossible as it would require to recompile the m2e code on the fly and reload your eclipse, that's why it is called **embedded** maven. You can only control the maven version used to **launch external** maven runs.

**Comment by tivervac** [comment]
IIUC, dependency resolution of an external maven run is done by the supplied maven then, right? That would make sense.

**Comment by laeubi** [comment]
As said if you choose `Run As > Maven` then m2e uses the configured maven installs (defaults to embedded), in all other cases it uses internal/embedded maven.

**Comment by tivervac** [comment]
Alright, that answers my question, thanks for the quick answer!
